cinema/views.py

```diff
@@ -43,3 +50,109 @@ def movie_detail(request, pk):
     if request.method == "DELETE":
         movie.delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+# <---
+
+
+class GenreList(APIView):
+    def get(self, request):
+        genres = Genre.objects.all()
+        serializer = GenreSerializer(genres, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def post(self, request):
+        serializer = GenreSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class GenreDetail(APIView):
+    def get_object(self, pk):
+        try:
+            return Genre.objects.get(pk=pk)
+        except Genre.DoesNotExist:
+            raise Http404
```

Comment:
using `get_object_or_404` would be better

Reply:
> using `get_object_or_404` would be better
I've used this shortcut in my previous tasks. Wanted to implement something else. It is not a mistake, I guess

Reply:
yes, that is not a mistake, just my advice. Don't think I'm being too picky about you, you just always have good code and I can't just keep approving it. I dig into the little things so that everything is perfect

Reply:
I really appreciate the passion you apply to code reviews :)